Automattic/wp-calypso
# Implement Google One Tap Login

## Proposed Changes

This adds inlined Google One Tap login hidden behind a flag. Context is here: peP6yB-29m-p2

**Requires** D155574-code [Deployed]

## Screenshots

### Before

![image](https://github.com/user-attachments/assets/fed9a730-18d2-4c9b-879b-37a6548f8a13)

### After
![image](https://github.com/user-attachments/assets/5a944f9b-936d-429f-afa8-643b8b67817f)

## Why are these changes being made?
To make signing in with Google smoother.

## Design challenges
Sadly, the Google button can't be resized. So I had to do some magic to make it fit our width. But that also means it's taller than the other buttons. We may resize the other buttons or keep them as-is.

## Locale issues
The Google library ignores both `hl` and `locale` configurations and uses the IP address as the locale. This will cover most cases but it's bad that it ignores Calypso locale. I spent an hour looking for a solution but couldn't find one. 

## Testing Instructions

This should be tested locally.

1. Checkout this branch.
2. Run `yarn start-mock-wordpress-com`.
3. Add `127.0.0.1 wordpress.com` to your hosts file.
4. Sandbox public-api.wordpress.com.
5. Sandbox de.wordpress.com.
6. Go to https://wordpress.com/start/user-social?flags=google_one_tap.
7. You should see One Tap Google Sign in.
